### ISSUE TITLE
TRZReader doesn't complain when wrong topology is provided

### ISSUE THREAD

**Issue by yuxuanzhuang** [MDAnalysis/mdanalysis#2817]
## Expected behavior ##

TRZReader raises a error (or warning) when a wrong topology (or n_atoms) is provided to `TRZReader`.

## Actual behavior ##

It passes without complaining during initialization.

## Code to reproduce the behavior ##

```python
u = mda.Universe(TRZ, n_atoms=8180)  # should be 8184
#  pass
u = mda.Universe(PSF, TRZ) #  should be TRZ_psf
#  pass
```
If you check for `dt` or `read_frame`, it will run into error. Maybe add such checks during initialization?
```python
>>> u.trajectory.dt
~/mdanalysis/package/MDAnalysis/coordinates/TRZ.py in _seek(self, nframes)
    387             seeksize = int(seeksize)
    388 
--> 389             self.trzfile.seek(seeksize, 1)
    390 
    391     def _reopen(self):

OSError: [Errno 22] Invalid argument

>>> u.trajectory[0]
~/mdanalysis/package/MDAnalysis/coordinates/base.py in _apply_limits(self, frame)
   1565             frame += len(self)
   1566         if frame < 0 or frame >= len(self):
-> 1567             raise IndexError("Index {} exceeds length of trajectory ({})."
   1568                              "".format(frame, len(self)))
   1569         return frame

IndexError: Index 0 exceeds length of trajectory (0).
```

Iteration, on the other hand, works--with wrong result.
```python
>>> u.trajectory.__next__()
< Timestep 1023497743 with unit cell dimensions [5.6476583e-04 1.2095231e-03 1.3170627e-02 1.0533933e+01 1.7983182e+02
 1.6946487e+02] >
```

## Current version of MDAnalysis ##

- Which version are you using? (run `python -c "import MDAnalysis as mda; print(mda.__version__)"`) 2.0.0.dev
- Which version of Python (`python -V`)? 3.8
- Which operating system? Ubuntu 20.04
